comment/comment.go

```diff
@@ -5,17 +5,20 @@ import (
 	"time"
 
 	"github.com/fabric8-services/fabric8-wit/gormsupport"
+	"github.com/fabric8-services/fabric8-wit/id"
+
 	uuid "github.com/satori/go.uuid"
 )
 
 // Comment describes a single comment
 type Comment struct {
 	gormsupport.Lifecycle
-	ID       uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
-	ParentID uuid.UUID `sql:"type:uuid"`
-	Creator  uuid.UUID `sql:"type:uuid"` // Belongs To Identity
-	Body     string
-	Markup   string
+	ID              uuid.UUID   `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	ParentID        uuid.UUID   `sql:"type:uuid"`
+	ParentCommentID id.NullUUID `sql:"type:uuid"`
```

Comment:
There should be a check that the `ParentCommentID` != `ID`

Reply:
Well, that is not possible as the UUID of the new comment is created when inserting into the db. So the client can't already know the UUID beforehand. On update requests, the ParentCommentID is ignored (like all other non-updateable fields).